optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/shift/ShiftRestServiceImpl.java

```diff
@@ -131,58 +114,24 @@ public Boolean removeShift(Integer tenantId, Long id) {
 
     @Override
     @Transactional
-    public List<Long> addShiftsFromTemplate(Integer tenantId,
-            String startDateString, String endDateString) throws Exception {
-        ShiftTemplate template = getTemplate(tenantId);
-
-        if (null == template) {
-            throw new IllegalStateException("You cannot add shifts if you don't have a template!");
+    public List<Long> addShiftsFromTemplate(Integer tenantId, Integer lengthInDays) {
+        Collection<ShiftTemplate> shiftTemplates = getTemplate(tenantId);
+
+        TenantConfiguration tenantConfiguration = entityManager.find(Tenant.class, tenantId).getConfiguration();
+        ShiftFileParser.ParserOut parserOutput = shiftFileParser.parse(tenantId, tenantConfiguration, null,
+                                                                       lengthInDays, shiftTemplates);
+
+        List<Shift> shifts = parserOutput.getShiftOutputList();
+        List<EmployeeAvailability> employeeAvailabilities = parserOutput.getEmployeeAvailabilityOutputList();
+        List<Long> out = new ArrayList<Long>();
```

Comment:
You can use Java 7's diamonds here

Reply:
Will update (and one day submit a PR to the Eclipse team to show these as warnings)

Reply:
Sure it doesn't have it? Maybe your language target is set to Java 1.6 or something like that? (it's been a while I don't work on Eclipse)